sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -377,6 +378,39 @@ case class InsertIntoTable(
   override lazy val resolved: Boolean = childrenResolved && table.resolved
 }
 
+/** Factory for constructing new `View` nodes. */
+object View {
+  def apply(desc: CatalogTable): View = View(desc, desc.schema.toAttributes, None)
+}
+
+/**
+ * A container for holding the view description(CatalogTable), and the output of the view. The
+ * child will be defined if the view is resolved with Hive support, else it should be None.
+ * This operator will be removed at the end of analysis stage.
+ *
+ * @param desc A view description(CatalogTable) that provides necessary information to resolve the
+ *             view.
+ * @param output The output of a view operator, this is generated during planning the view, so that
+ *               we are able to decouple the output from the underlying structure.
+ * @param child The logical plan of a view operator, it should be non-empty if the view is resolved
+ *              with Hive support, else it should be None.
+ */
+case class View(
+    desc: CatalogTable,
+    output: Seq[Attribute],
+    child: Option[LogicalPlan] = None) extends LogicalPlan with MultiInstanceRelation {
```

Comment:
Please support the InMemory catalog as well. IMO the only argument to keep the option would be that we want to do the parsing in the analyzer (that could give you more control over the parsing process). I am fine with both.